src/wp-includes/post.php

```diff
@@ -5674,52 +5672,41 @@ function get_page( $page, $output = OBJECT, $filter = 'raw' ) {
  * @return WP_Post|array|null WP_Post (or array) on success, or null on failure.
  */
 function get_page_by_path( $page_path, $output = OBJECT, $post_type = 'page' ) {
-	global $wpdb;
-
-	$last_changed = wp_cache_get_last_changed( 'posts' );
-
-	$hash      = md5( $page_path . serialize( $post_type ) );
-	$cache_key = "get_page_by_path:$hash:$last_changed";
-	$cached    = wp_cache_get( $cache_key, 'posts' );
-	if ( false !== $cached ) {
-		// Special case: '0' is a bad `$page_path`.
-		if ( '0' === $cached || 0 === $cached ) {
-			return;
-		} else {
-			return get_post( $cached, $output );
-		}
-	}
-
-	$page_path     = rawurlencode( urldecode( $page_path ) );
-	$page_path     = str_replace( '%2F', '/', $page_path );
-	$page_path     = str_replace( '%20', ' ', $page_path );
-	$parts         = explode( '/', trim( $page_path, '/' ) );
-	$parts         = array_map( 'sanitize_title_for_query', $parts );
-	$escaped_parts = esc_sql( $parts );
-
-	$in_string = "'" . implode( "','", $escaped_parts ) . "'";
+	$page_path = rawurlencode( urldecode( $page_path ) );
+	$page_path = str_replace( '%2F', '/', $page_path );
+	$page_path = str_replace( '%20', ' ', $page_path );
+	$parts     = explode( '/', trim( $page_path, '/' ) );
+	$parts     = array_map( 'sanitize_title_for_query', $parts );
 
 	if ( is_array( $post_type ) ) {
 		$post_types = $post_type;
 	} else {
 		$post_types = array( $post_type, 'attachment' );
 	}
 
-	$post_types          = esc_sql( $post_types );
-	$post_type_in_string = "'" . implode( "','", $post_types ) . "'";
-	$sql                 = "
-		SELECT ID, post_name, post_parent, post_type
-		FROM $wpdb->posts
-		WHERE post_name IN ($in_string)
-		AND post_type IN ($post_type_in_string)
-	";
+	$args = array(
+		'post_name__in'          => $parts,
+		'post_type'              => $post_types,
+		'post_status'            => 'all',
```

Comment:
@peterwilsoncc Fun fact this work!
<img width="1120" alt="Screenshot 2022-12-13 at 12 53 17" src="https://user-images.githubusercontent.com/237508/207323374-852d9e59-b015-4056-9cbc-c18c8ccee344.png">

Reply:
Possibly a manifestation of https://core.trac.wordpress.org/ticket/51094

Reply:
@peterwilsoncc How about adding a new special case for post status, where if you are give `all`, it returns all. This maybe confusing with `any`. But it clear to me that we need a state to get all posts of any post status, there is at least two places in core that need it.

Reply:
We could also do the same workaround as in https://github.com/WordPress/wordpress-develop/commit/7ad9e7d04e70c1f63c024ef58065aa49ca30eab0.